### DISCUSSION TITLE
Snap package support

### DISCUSSION THREAD

**Discussion by boukendesho** [Jeffser/Alpaca#302] [General]
Hi,  👋
I took a quick look at your snapcraft.yaml.

> but it doesn't show the app on the app menu

Desktop app needs a special meta param called 'desktop' , see [here](https://snapcraft.io/docs/desktop-menu-support)
Also you need to specify the Icon file, put it under snap/gui/ directory. File structure will like this:
$ tree .
snap
|
- snapcraft.yaml
- gui
    |
    - _your_desktop_file_.desktop
    - _your_icon_.png/svg

For better cli experiences, i suggest you add one more entry under **apps** in snapcraft.yaml , called ollama, to expose the ollama cli .
After that, you can run it from the terminal, like this  $ alpaca.ollama run llama3.1

I'll continue with the rest and if I have time, then will test and submit a PR.

Cheers!

**Comment by boukendesho** [comment]
Hi, i have one question. 

How your program commuicate with ollama, through http request?

Do we need run `ollama serve` in the background?

**Reply by Jeffser** [reply]
Yes the app runs the command `ollama serve` and it communicates using http

**Comment by boukendesho** [comment]
![Screenshot from 2024-09-16 14-57-09](https://github.com/user-attachments/assets/888e22b8-40e9-439b-9226-2ef02c6a352f)
Just got first runnable snap version !

But there's still a lot of things that need to be perfected.

TODO:
1. Ollama with dGPU test. (i don't have a dGPU )
2. Snap plugs tuning
3. Ollama overrides test.
4. Make a option(such as env parameters) that can let ollama read model files from removable media.
5. Arm64 build support.
6. More tests.

My internet is not so good and very slow to download the models, these are what I can do for now. 
Cheers.

**Comment by boukendesho** [comment]
One more issue, downloaded models through gui didn't show in cli results.
$ alpaca.ollama list   show nothing.

I found models are store in ~/snap/alpaca/current/.local/share/com.jeffser.Alpaca/.ollama/, which is not the default path for ollama, may be the key.

**Reply by Jeffser** [reply]
> One more issue, downloaded models through gui didn't show in cli results. $ alpaca.ollama list show nothing.

Yeah it's because of the different directory, I'll see what I can do

**Comment by Jeffser** [comment]
thanks for your help!
